From 82ff2b6a12ff4e98d946605f622634ca1a806cd9 Mon Sep 17 00:00:00 2001
From: "Koeller, Thomas" <Thomas.Koeller@baslerweb.com>
Date: Fri, 11 Dec 2020 15:19:36 +0100
Subject: [PATCH 17/20] [imx8mq] arm64: dts: imx8mq: Added highspeed device
 tree for imx8mq

Added highspeed device tree for imx8mq

This device tree allows for full-speed CSI operation at 333MHz clock
frequency.
---
 ...imx8mq-evk-basler-camera-csi-fullspeed.dts | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mq-evk-basler-camera-csi-fullspeed.dts

diff --git a/arch/arm64/boot/dts/freescale/imx8mq-evk-basler-camera-csi-fullspeed.dts b/arch/arm64/boot/dts/freescale/imx8mq-evk-basler-camera-csi-fullspeed.dts
new file mode 100644
index 000000000000..388a90630cb9
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mq-evk-basler-camera-csi-fullspeed.dts
@@ -0,0 +1,36 @@
+#include "imx8mq-evk.dts"
+#include "imx8mq-evk-basler-camera-csi.dtsi"
+
+&csi1_bridge {
+	status = "okay";
+};
+
+&csi2_bridge {
+	status = "okay";
+};
+
+&mipi_csi_1 {
+	status = "okay";
+	assigned-clock-rates = <266000000>, <333000000>, <66000000>;
+};
+
+&mipi_csi_2 {
+	status = "okay";
+	assigned-clock-rates = <266000000>, <333000000>, <66000000>;
+};
+
+&basler_camera_mipi2 {
+	status = "okay";
+};
+
+&basler_camera_mipi1 {
+	status = "okay";
+};
+
+&i2c1 {
+	link-frequencies = /bits/ 64 <600000000>;
+};
+
+&i2c2 {
+	link-frequencies = /bits/ 64 <600000000>;
+};
-- 
2.34.1

